From c3ee8421d56c14eaef2029427864237826a0d7d9 Mon Sep 17 00:00:00 2001
From: Panu Matilainen <pmatilai@redhat.com>
Date: Thu, 20 Nov 2014 12:27:23 -0500
Subject: [net] ipv6: drop unused fib6_clean_all_ro() function and rt6_proc_arg struct

Message-id: <6b61340e8a900eff2cca5aade82dfcd0f4cc29e3.1416484884.git.pmatilai@redhat.com>
Patchwork-id: 100386
O-Subject: [RHEL7 PATCH net 2/2] ipv6: drop unused fib6_clean_all_ro() function and rt6_proc_arg struct
Bugzilla: 1163605
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Marcelo Leitner <mleitner@redhat.com>
RH-Acked-by: Daniel Borkmann <dborkman@redhat.com>

Straight backport of removing now unused fib6_clean_all_ro() and
never used rt6_proc_arg struct. Does not affect any functionality.

Second half of:
commit 8d2ca1d7b5c3e63b3a8a81ae99015b615c5f2bf7
Author: Hannes Frederic Sowa <hannes@stressinduktion.org>
Date:   Sat Sep 21 16:55:59 2013 +0200

    [...]

    Also kill never used struct rt6_proc_arg and now unused function
    fib6_clean_all_ro.

    Cc: Ben Greear <greearb@candelatech.com>
    Cc: Patrick McHardy <kaber@trash.net>
    Cc: YOSHIFUJI Hideaki <yoshfuji@linux-ipv6.org>
    Signed-off-by: Hannes Frederic Sowa <hannes@stressinduktion.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Panu Matilainen <pmatilai@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/net/ip6_fib.h b/include/net/ip6_fib.h
index c789378..46fbf3d 100644
--- a/include/net/ip6_fib.h
+++ b/include/net/ip6_fib.h
@@ -286,10 +286,6 @@ struct fib6_node  *fib6_locate(struct fib6_node *root,
           const struct in6_addr *daddr, int dst_len,
           const struct in6_addr *saddr, int src_len);
 
-extern void   fib6_clean_all_ro(struct net *net,
-            int (*func)(struct rt6_info *, void *arg),
-            int prune, void *arg);
-
 extern void   fib6_clean_all(struct net *net,
             int (*func)(struct rt6_info *, void *arg),
             void *arg);
diff --git a/net/ipv6/ip6_fib.c b/net/ipv6/ip6_fib.c
index d948b97..32eccfd 100644
--- a/net/ipv6/ip6_fib.c
+++ b/net/ipv6/ip6_fib.c
@@ -1529,25 +1529,6 @@ static void fib6_clean_tree(struct net *net, struct fib6_node *root,
  fib6_walk(&c.w);
 }
 
-void fib6_clean_all_ro(struct net *net, int (*func)(struct rt6_info *, void *arg),
-      int prune, void *arg)
-{
- struct fib6_table *table;
- struct hlist_head *head;
- unsigned int h;
-
- rcu_read_lock();
- for (h = 0; h < FIB6_TABLE_HASHSZ; h++) {
-  head = &net->ipv6.fib_table_hash[h];
-  hlist_for_each_entry_rcu(table, head, tb6_hlist) {
-   read_lock_bh(&table->tb6_lock);
-   fib6_clean_tree(net, &table->tb6_root,
-     func, prune, arg);
-   read_unlock_bh(&table->tb6_lock);
-  }
- }
- rcu_read_unlock();
-}
 void fib6_clean_all(struct net *net, int (*func)(struct rt6_info *, void *arg),
       void *arg)
 {
diff --git a/net/ipv6/route.c b/net/ipv6/route.c
index a3f7ad1..44f77ce 100644
--- a/net/ipv6/route.c
+++ b/net/ipv6/route.c
@@ -2829,15 +2829,6 @@ static int ip6_route_dev_notify(struct notifier_block *this,
 
 #ifdef CONFIG_PROC_FS
 
-struct rt6_proc_arg
-{
- char *buffer;
- int offset;
- int length;
- int skip;
- int len;
-};
-
 static const struct file_operations ipv6_route_proc_fops = {
  .owner  = THIS_MODULE,
  .open  = ipv6_route_open,
-- 
1.7.1